delta-lake/src/main/java/org/apache/iceberg/delta/BaseMigrateDeltaLakeTableAction.java

```diff
@@ -0,0 +1,356 @@
+        totalDataFiles,
+        deltaTableLocation);
+    return new BaseMigrateDeltaLakeTableActionResult(totalDataFiles);
+  }
+
+  private Schema convertDeltaLakeSchema(io.delta.standalone.types.StructType deltaSchema) {
+    Type converted =
+        DeltaLakeDataTypeVisitor.visit(deltaSchema, new DeltaLakeTypeToType(deltaSchema));
+    return new Schema(converted.asNestedType().asStructType().fields());
+  }
+
+  private PartitionSpec getPartitionSpecFromDeltaSnapshot(Schema schema) {
+    List<String> partitionNames = deltaLog.snapshot().getMetadata().getPartitionColumns();
+    if (partitionNames.isEmpty()) {
+      return PartitionSpec.unpartitioned();
+    }
+
+    PartitionSpec.Builder builder = PartitionSpec.builderFor(schema);
+    for (String partitionName : partitionNames) {
+      builder.identity(partitionName);
+    }
+    return builder.build();
+  }
+
+  private void copyFromDeltaLakeToIceberg(Transaction transaction) {
+    Iterator<VersionLog> versionLogIterator =
+        deltaLog.getChanges(
+            0, // retrieve actions starting from the initial version
+            false); // not throw exception when data loss detected
+
+    while (versionLogIterator.hasNext()) {
+      VersionLog versionLog = versionLogIterator.next();
+      commitDeltaVersionLogToIcebergTransaction(versionLog, transaction);
+    }
+  }
+
+  private void commitDeltaVersionLogToIcebergTransaction(
+      VersionLog versionLog, Transaction transaction) {
+    List<Action> actions = versionLog.getActions();
+
+    // We first need to iterate through to see what kind of transaction this was. There are 3
+    // cases:
+    // 1. AppendFile - when there are only AddFile instances (an INSERT on the table)
+    // 2. DeleteFiles - when there are only RemoveFile instances (a DELETE where all the records
+    // of file(s) were removed
+    // 3. OverwriteFiles - when there are a mix of AddFile and RemoveFile (a DELETE/UPDATE)
+
+    // Create a map of Delta Lake Action (AddFile, RemoveFile, etc.) --> List<Action>
+    Map<String, List<Action>> deltaLakeActionMap =
+        actions.stream()
+            .filter(action -> action instanceof AddFile || action instanceof RemoveFile)
+            .collect(Collectors.groupingBy(a -> a.getClass().getSimpleName()));
+
+    List<DataFile> filesToAdd = Lists.newArrayList();
+    List<DataFile> filesToRemove = Lists.newArrayList();
+    for (Action action : Iterables.concat(deltaLakeActionMap.values())) {
+      DataFile dataFile = buildDataFileFromAction(action, transaction.table());
+      if (action instanceof AddFile) {
+        filesToAdd.add(dataFile);
+      } else if (action instanceof RemoveFile) {
+        filesToRemove.add(dataFile);
+      } else {
+        throw new ValidationException(
+            "The action %s's is unsupported", action.getClass().getSimpleName());
+      }
+    }
+
+    if (filesToAdd.size() > 0 && filesToRemove.size() > 0) {
+      // Overwrite_Files case
```

Comment:
nit: why _ between the `Overwrite_Files`? It does not match the comments on L201